lib/bk/commands/artifacts.rb

```diff
@@ -137,23 +135,24 @@ def glob_matches?(glob, path)
         end
       end
 
-      def download_artifact(artifact)
+      def download_artifact(artifact, header, bar)
         path = Pathname.new("tmp/bk/#{artifact.path}")
         if path.exist?
-          puts "#{path} already exists, skipping"
+          bar.log "#{header}: #{path} already exists, skipping"
           return
         end
 
         sleep_duration = 1
         begin
+          bar.log "#{header}: Downloading artifact to tmp/bk/#{artifact.path}"
           download_url = artifact.to_h["downloadURL"]
           redirected_response_from_aws = Net::HTTP.get_response(URI(download_url))
           artifact_response = Net::HTTP.get_response(URI(redirected_response_from_aws["location"]))
           FileUtils.mkdir_p(path.dirname)
           path.write(artifact_response.body)
-        rescue
+        rescue => ex
           return if sleep_duration > 300
-          puts "Retry"
+          bar.log "Retrying after #{sleep_duration} seconds (encountered error: #{ex})"
```

Comment:
Example:

I had Zscaler disconnected and got:
```
Retrying after 32 seconds (encountered error: SSL_connect returned=1 errno=0 peeraddr=100.64.1.2:443 state=error: unexpected eof while reading)
```

I wanted to see (A) where we are in the exponential retry backoff loop and (B) what the error was.